### PR TITLE
Adds martial art datum and three styles

### PR DESCRIPTION
**Pull request by AnturK**
Humans hand-to-hand only.
Right now only boxing is obtainable without badminnery by putting on boxing gloves.
##### Boxing:
- No Grabbing
- No Disarming
- Punches always hit with 5-8 Stamina Damage
- If the defender has more than 50 Stamina damage each hit has stamina_damage +- 15% chance to knockout
##### Wrestling:
- Harm intent has 50% chance to jump straight to aggressive grab
- Disarm on a grabbed person does a suplex stunning both (person getting attacked moment longer) and doing 30 Brute to the defender.
- If there's no grabbed person it does the same thing as harm intent
- Grab does standard grab + 10 Stamina Damage
##### Plasma Fist:

Each hit works as default harm attack and builds up a streak if it's used on the same person
Harm Harm Disarm - Torando - Basically wizard's repulse around the user
Disarm Harm Disarm - Knockback - Throws the defender from the user
HDDDH - Plasma Fist - gibs the defender leaving the brain.
##### Other stuff:
- Adds Plasma Fist Scroll and Wrestling Belt - both badmin only
- Adds on_unequip hook for items. <- This is probably the biggest problem here since i'm not entirely sure this will be called in every case.

I meant to add traitor buyable style or sec training one but these things are harder to balance than they seem.. 
Possible expansion : make these handle hand to hand defense too

### CODE REVIEW

**Comment by AlexCrimson** [comment]
The wrestling one seems useful for lings. They could use some better grab mechanics.

**Comment by Miauw62** [comment]
This sounds absolutely hilarious.
Op 7-apr.-2015 20:12 schreef "AlexCrimson" <notifications@github.com>:

**Comment by Alek2ander** [comment]
:+1:
Styles should also have triggers on movement (both you and adjacent enemies). Because power slides. And tripping. And a lot of other fun things that can be done here.

**Comment by Iamgoofball** [comment]
WRESTLING BELT TRAITOR ITEM WHEN

:+1: X :100:

**Comment by AnturK** [comment]
Yeah there's a lot of additional hooks you could add but wanted to start with something basic for now.

**Comment by Iamgoofball** [comment]
snap into a fucking slim jim

**Comment by RemieRichards** [comment]
@AnturK 
>Adds on_unequip hook for items. <- This is probably the biggest problem here since i'm not entirely sure this will be called in every case.

```dropped()``` is called when the object is unequipped from an inventory slot, source: I once added an unequip hook myself, only to find it's functionality entirely eclipsed by dropped().

**Comment by AnturK** [comment]
Ah, alright i'll switch it to that then. Dropped could use a comment about that i guess

**Comment by RemieRichards** [comment]
@AnturK it probably could yeah, My guess is it was originally just it's name sake, "dropped" but ended up covering the functionality of an unequip hook, and nobody bothered to document it.

File: code/datums/martial.dm [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	var/atom/throw_target = get_edge_target_turf(D, get_dir(D, get_step_away(D, A)))
+	D.throw_at(throw_target, 200, 4)
+	A.say("HYAH!")
+	return
+
+/datum/martial_art/plasma_fist/proc/Plasma(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	A.do_attack_animation(D)
+	playsound(D.loc, 'sound/weapons/punch1.ogg', 50, 1, -1)
+	A.say("PLASMA FIST!")
+	D.visible_message("<span class='danger'>[A] has hit [D] with PLASMA FIST TECHNIQUE!</span>", \
```
Comment (RemieRichards) [inline code/datums/martial.dm]: since PLASMA FIST is named, it should be THE PLASMA FIST TECHNIQUE.

File: code/datums/martial.dm [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+	if(!ishuman(user) || used)
+		return
+	var/mob/living/carbon/human/H = user
+	H.martial_art = new/datum/martial_art/plasma_fist
+	used = 1
+	desc += "It looks like it's magic was used up."
+
+/datum/martial_art/plasma_fist
+	name = "Plasma Fist"
+	var/streak = ""
```
Comment (RemieRichards) [inline code/datums/martial.dm]: mind moving this (and any other "unique functionality" variables/procs) down to the base class?
So we can easily have other martial_arts do streaks, etc.

**Comment by RemieRichards** [comment]
Also since I havent actually said, I like this :+1:

**Comment by AlexCrimson** [comment]
Should Ninja and Deathsquads get their own style? Something stupidly OP or gimmicky?

**Comment by RemieRichards** [comment]
Oh hey guess whats coming in the next Ninja PR now then... Hehe

File: code/modules/mob/living/carbon/human/species.dm [hunk 3]
```diff
@@ -750,107 +752,115 @@
...
-						talked = 1
-					spawn(1)
-						qdel(lgrab)
-				if(istype(H.r_hand, /obj/item/weapon/grab))
-					var/obj/item/weapon/grab/rgrab = H.r_hand
-					if(rgrab.affecting)
-						H.visible_message("<span class='warning'>[M] has broken [H]'s grip on [rgrab.affecting]!</span>")
+			if(attacker_style && attacker_style.disarm_act(M,H))
+				return 1
+			else:
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/species.dm]: "else:"

what's with the colon?

Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species.dm]: iirc @AnturK is working on other shit in other languages.
I know python uses colons and such, so it's probably just force of habit.

**Comment by Midaychi** [comment]
Krava-Naga spoof for nook ops and deathsquids, that focuses on quick takedowns and weapon disarm+obtaining.

**Comment by RemieRichards** [comment]
@Midaychi you mean Krav-maga?

**Comment by Lumipharon** [comment]
Fucking fantastic.

**Comment by Iamgoofball** [comment]
so fucking hyped for this shit

@Midaychi
nah they need to be able to buy wrestling belts

wrassler ops 2K15

File: code/modules/mob/living/carbon/human/species.dm (hunk 3 above)
Comment (AnturK) [inline code/modules/mob/living/carbon/human/species.dm]: Goddamit i did it again. And yeah i work with python at my job

Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/species.dm]: I python for procrastinating by making Buttbot IRCbots and other misc terrible scripts.

**Comment by Midaychi** [comment]
Krav-naga would be the spoof.

**Comment by Fox-McCloud** [comment]
Fox McCloud screams!

Fox McCloud climbs up onto the table!

Fox McCloud elbow drops Griffer McFartface

Griffer McFartface's arm flies off in an arc!

File: code/datums/martial.dm [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+		D.apply_effect(4, WEAKEN, armor_block)
+		D.forcesay(hit_appends)
+	else if(D.lying)
+		D.forcesay(hit_appends)
+	return
+
+/datum/martial_art/plasma_fist/proc/Tornado(var/mob/living/carbon/human/A, var/mob/living/carbon/human/D)
+	A.say("TORNADO SWEEP!")
+	spawn(0)
+		for(var/i in list(1,2,4,8,4,2,1,2,4,8,4,2))
```
Comment (optimumtact) [inline code/datums/martial.dm]: Please use defines for this

Comment (AnturK) [inline code/datums/martial.dm]: Yeah, just grabbed this straight from esword code, will do

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 5]
```diff
@@ -46,4 +46,6 @@
 	var/gender_ambiguous = 0 //if something goes wrong during gender reassignment this generates a line in examine
 
 	var/blood_max = 0 //how much are we bleeding
-	var/bleedsuppress = 0 //for stopping bloodloss, eventually this will be limb-based like bleeding
\ No newline at end of file
+	var/bleedsuppress = 0 //for stopping bloodloss, eventually this will be limb-based like bleeding
+
+	var/datum/martial_art/martial_art = null
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/human/human_defines.dm]: We may as well apply this at the mob/living level, so things like aliens/monkeys can have their own fighting techniques

**Comment by optimumtact** [comment]
Nice and modular :+1:

**Comment by Lumipharon** [comment]
secret monkey technique.

**Comment by AlexCrimson** [comment]
Can we have Bruce Lee Enter the Dragon punch/kick sounds?

**Comment by PKPenguin321** [comment]
would this enable the idea of brass knuckles for traitors that increase punch damage?

**Comment by MrPerson** [comment]
This seems like a goofy gimmick that goon would add, not us.

**Comment by dannno** [comment]
I think it could be made serious, doesn't have to be gimmicky
the only issue is balancing, making sure that one style isn't better than the others

**Comment by TZK13** [comment]
Can you make it so that you can't kick people while wearing boxing gloves? I've always desired this for more fair fights.

File: code/modules/mob/living/carbon/human/human_defines.dm (hunk 5 above)
Comment (AnturK) [inline code/modules/mob/living/carbon/human/human_defines.dm]: I was considering this but then again animals are not known for varied fighting and many critters already have unique attack_whatever defined so subtypes overriding this seemed enough. I can see how much i'd need to move around and try doing this later today maybe.

**Comment by Miauw62** [comment]
Add a tableclimbing hook so wrestlers can supplex people :^]
Op 8-apr.-2015 10:13 schreef "TZK13" <notifications@github.com>:

**Comment by Supermichael777** [comment]
THE RAGE CAGE IS REAL

**Comment by SamCroswell** [comment]
What happens if someone is wearing both boxing gloves *and* a wrestling belt?
Does their fighting style evolve into the legendary macho-boxing style?

**Pull request merged**